ocs_ci/utility/ibmcloud.py

```diff
@@ -203,3 +205,232 @@ def add_deployment_dependencies():
+                f"  --attachment {attachment_id}"
+            )
+            out = run_cmd(cmd)
+            logger.info(f"detachment command output: {out}")
+
+    def get_node_by_attached_volume(self, volume):
+        """
+        Get the node by attached volume on IBM Cloud.
+
+        Args:
+            volume (str): volume id.
+
+        """
+        cmd = f"ibmcloud is volume {volume} --output json"
+        out = run_cmd(cmd)
+        out = json.loads(out)
+
+        if not out["volume_attachments"]:
+            logger.info("volume is not attached to node")
+            raise NodeHasNoAttachedVolume("not attached to node")
+        else:
+            worker_id = out["volume_attachments"][0]["instance"]["name"]
+            logger.info(f"volume is  attached to node: {worker_id}")
+            return worker_id
+
+    def get_data_volumes(self):
+        """
+        Get volumes in IBM Cloud.
+
+        """
+        logger.info("get data volumes")
+
+        cmd = "ibmcloud is vols --output json"
+        out = run_cmd(cmd)
+        out = json.loads(out)
+
+        vol_ids = []
+        for vols in out:
+            vol_ids.append(vols["id"])
+
+        logger.info(f"volume ids are : {vol_ids}")
+        return vol_ids
+
+    def wait_for_volume_attach(self, volume):
+        """
+        Checks volume is attached to node or not
+
+        Args:
+            volume (Volume): The volume to wait for to be attached
+
+        Returns:
+            bool: True if the volume has been attached to the
+                instance, False otherwise
+
+        """
+        cmd = f"ibmcloud is volume {volume} --output json"
+        out = run_cmd(cmd)
+        out = json.loads(out)
+
+        if not out["volume_attachments"]:
+            logger.info("volume is not attached to node")
+            return False
+        else:
+            logger.info("volume is attached to node")
+            return True
+
+    def getVolumeId(self):
+        """
+        Returns Volumeid with the name taken from constants
+
+        Returns:
+            str: volume id if the volume exists otherwise create
+                new volume
+
+        """
+        zone = config.ENV_DATA["zone"]
+
+        cmd = "ibmcloud is vols --output json "
+        out = run_cmd(cmd)
+        out = json.loads(out)
+
+        vol_id = ""
+        for vols in out:
+            vol_name = str(vols["name"]).strip()
+            if vol_name == constants.IBMCLOUD_VOLUME_NAME:
+                vol_id = vols["id"]
+                return vol_id
+
+        if len(vol_id) == 0:
+            cmd = f"ibmcloud is volume-create {constants.IBMCLOUD_VOLUME_NAME} general-purpose {zone} --output json"
+            out = run_cmd(cmd)
+            out = json.loads(out)
+            vol_id = out["id"]
+            return vol_id
+
+    def restart_nodes_by_stop_and_start(self, nodes, force=True):
+        """
+        Reboot the node on IBM Cloud.
+
+        """
```

Comment:
Does this function really reboot the nodes by stopping and then starting them?

Also based on the logic, it seems like it reboot only subset of the nodes which are not ready, please update the docstring accordingly.

Reply:
Done